src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.tsx

```diff
@@ -15,43 +24,181 @@ const ContactTagChip = styled(Chip)(({ theme }) => ({
+                  tagList: updateData?.updateContact?.contact.tagList,
+                },
+              };
+              cache.writeQuery({ ...query, data });
+              enqueueSnackbar(t('Tag successfully removed'), {
+                variant: 'success',
+              });
+            }
+          },
+        });
+      } catch (error) {
+        enqueueSnackbar(error.message, { variant: 'error' });
+        throw error;
+      }
     }
-    console.info('You clicked the Delete tag for tag' + tag + '' + contactId);
   };
-  const handleAddTag = () => {
-    // TODO: add mutation to update tag for contact
-    console.info('You clicked add contact Tag to contact' + contactId);
+
+  const onSubmit = async ({ tag }: { tag: string }) => {
+    try {
+      updateContactTags({
+        variables: {
+          accountListId,
+          contactId,
+          tagList: [...contactTags, tag],
+        },
+        update: (cache, { data: updateData }) => {
+          const tagList = updateData?.updateContact?.contact.tagList;
+          const query = {
+            query: ContactDetailsTabDocument,
+            variables: {
+              accountListId,
+              contactId,
+            },
+          };
+          const dataFromCache = cache.readQuery<ContactDetailsTabQuery>(query);
+
+          if (dataFromCache) {
+            const data = {
+              contact: {
+                ...dataFromCache.contact,
+                tagList,
+              },
+            };
+            cache.writeQuery({ ...query, data });
+            enqueueSnackbar(t('Tag successfully added'), {
+              variant: 'success',
+            });
+          }
+        },
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
   };
 
   return (
-    <ContactTagsContainer>
+    <ContactTagsContainer display="flex" alignItems="center">
+      <ContactTagIcon />
       {contactTags.map((tag) => (
-        <ContactTagChip key={tag} label={tag} onDelete={handleTagDelete} />
+        <ContactTagChip
+          key={tag}
+          label={tag}
+          disabled={updating}
+          onDelete={() => handleTagDelete(tag)}
+          title={t('Delete Icon')}
+        />
       ))}
-      <Link color="textSecondary" onClick={handleAddTag}>
-        <ContactTagAddTagText variant="subtitle1">
-          {t('add tag')}
-        </ContactTagAddTagText>
+      <Link color="textSecondary">
+        <Formik
+          initialValues={{ tag: '' }}
+          validationSchema={tagSchema}
+          onSubmit={(values, { setSubmitting, resetForm }) => {
+            onSubmit(values);
+            resetForm();
+            setSubmitting(false);
+          }}
+        >
+          {({
+            values: { tag },
+            handleChange,
+            handleSubmit,
+            isSubmitting,
+          }): ReactElement => (
+            <form onSubmit={handleSubmit} noValidate>
+              <ContactTagInput
```

Comment:
I'd appreciate it if you guys took a look at the storybook and let me know what you think of the design. In figma there's a design for the ```add tag``` text, but no design for actually adding a tag, so I improvised 🙂 I'm pretty happy with the results, but would like to hear others' feedback.

Reply:
Here's the link to the specific [storybook](https://5f112cf86767b10022b5b991-pfwkalpnmq.chromatic.com/?path=/story/contacts-tab-contactdetailstab-contacttags--default). But for some reason the styling is different in the storybook... not sure why 🤷‍♂️ Here's a picture of it actually in the app:
<img width="406" alt="Screen Shot 2021-05-27 at 4 04 27 PM" src="https://user-images.githubusercontent.com/39680460/119890589-07699100-bf06-11eb-9caa-770a06f0f5cd.png">
<img width="440" alt="Screen Shot 2021-05-27 at 4 04 34 PM" src="https://user-images.githubusercontent.com/39680460/119890594-08022780-bf06-11eb-9070-448dcf3c6ad0.png">

Reply:
Looks good to me. I haven't looked at storybook but can if you're still struggling with it there.